lms/services/canvas_api/factory.py

```diff
@@ -12,16 +12,21 @@ def canvas_api_client_factory(_context, request):
     :param request: Pyramid request object
     :return: An instance of CanvasAPIClient
     """
-    ai_getter = request.find_service(name="ai_getter")
+    application_instance_service = request.find_service(name="application_instance")
+    application_instance = application_instance_service.get()
+
+    canvas_host = urlparse(application_instance.lms_url).netloc
```

Comment:
Presumably this would be replaced with your new function?

Reply:
Yep, when a later PR adds the new function